### ISSUE TITLE
Docker + Poetry installation problem for Poetry=^1.2.0

### ISSUE THREAD

**Issue by NumberChiffre** [python-poetry/poetry#7416]
<!--
    Hi there! Thank you for submitting a bug report!

    Before you submit your issue, please review and follow the instructions at
    https://python-poetry.org/docs/contributing/#reporting-bugs. Fully fill out the template below; insufficient
    information or bad reproduction instructions will impair the ability of others to help you.
-->

<!-- All the below information must be provided for others to understand and help with your issue. -->
- **Poetry version**: 1.2.0
- **Python version**: 3.10.4
- **OS version and name**: Ubuntu 20.04
- **pyproject.toml**: NA

<!-- All the below steps should be completed before submitting your issue. Checked checkbox should look like this: [x] -->
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) stable Poetry version, installed using a recommended method.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have consulted the [FAQ](https://python-poetry.org/docs/faq/) and [blog](https://python-poetry.org/blog/) for any relevant entries or release notes.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option) and have included the output below.

## Issue
<!-- Now feel free to write your issue, and please be as descriptive as possible! Make sure to include detailed reproduction steps. -->
<!-- Thanks again 🙌 ❤ -->
Using my Dockerfile, I have the following:
```
FROM nvidia/cuda:12.0.0-base-ubuntu20.04 as base

# ****************Skipping steps to install packages*******************

RUN curl -sSL https://install.python-poetry.org | POETRY_HOME=/opt/poetry python && \
    cd /usr/local/bin && \
    ln -s /opt/poetry/bin/poetry

COPY poetry.lock poetry.lock
COPY poetry.toml poetry.toml
COPY pyproject.toml pyproject.toml

ARG INSTALL_DEV=true
RUN bash -c "if [ $INSTALL_DEV == 'true' ] ; then  \
    poetry install --no-root ; else  \
    poetry install --no-root --no-dev ; fi"
```

It seems that with:
```
[virtualenvs]
create = false
```

I'm getting the following error for poetry version >= 1.2.0:
```
 => ERROR [dev 1/1] RUN bash -c "if [ true == 'true' ] ; then      poetry install --no-root ; else      poetry install --no-root --no-dev ; fi"                                                                                                                               0.9s 
------                                                                                                                                                                                                                                                                             
 > [dev 1/1] RUN bash -c "if [ true == 'true' ] ; then      poetry install --no-root ; else      poetry install --no-root --no-dev ; fi":                                                                                                                                          
#24 0.577 Skipping virtualenv creation, as specified in config file.
#24 0.787 
#24 0.787 Invalid PEP 440 version: '0.23ubuntu1'
```

Current solution is to use poetry version == 1.1.15 to get around this. Would appreciate if you could point out what's wrong with my setup, or is this a recent bug (as I never had to downgrade my poetry version for docker build).

**Comment by finswimmer** [comment]
Hey,

this is a duplicate of #6334 and others. You want to use a venv - even in a docker container - to not get any conflicts between your project and the system.

fin swimmer
